Code/autopkg

```diff
@@ -2683,12 +2683,31 @@ def new_recipe(argv):
                 yaml.dump(recipe, f, encoding="utf-8")
         else:
             with open(filename, "wb") as f:
-                plistlib.dump(recipe, f)
+                plistlib.dump(plist_serializer(recipe), f)
         log(f"Saved new recipe to {filename}")
     except Exception as err:
         log_err(f"Failed to write recipe: {err}")
 
 
+def plist_serializer(obj):
+    """Serialize an object to ensure it can be dumped in plist format.
+
+    Args:
+        obj (dict, list): Object is assumed to be either a dict or list
+            that will be parsed.
+
+    Returns:
+        (any): The received object will be returned, modified if required.
+    """
+    if isinstance(obj, dict):
+        for k, v in obj.items():
+            obj[k] = "" if v is None else plist_serializer(v)
```

Comment:
This feels slightly dangerous and could cause type safety violations down the line, but I don't have a better answer to this now.

Reply:
Are you specifically referring to `isinstance(obj, dict)`
or `"" if v is None else plist_serializer(v)`?

I do know the former is a debate whether or not it should be used, but in this case, I as well, cannot think of another option at the moment.  I know the original object `run_results` being sent to the function, is a list of objects, but i was attempting to not assume what it would always be (when used by other functions; i.e. attempting to keep it agnostic).

For the latter, I've also tried to think of what could be fouled up with that condition, but haven't been able to think of anything yet.

Reply:
I was more concerned about the fact that any None value is inherently casted as an empty string instead. This is probably safe enough in our usage here, but as we add more and more type safety checks to AutoPkg's python code, this may need to be refactored a bit.

However, that is very much a "after Tomorrow" problem.

Reply:
Ah, I see.  I was thinking about that too, but I couldn't think of a way to otherwise infer what the value _should be_ to properly cast it into the corresponding plist/xml format.